hashicorp/consul
# [CC-7044] Start HCP manager as part of link creation

### Description
Depends on https://github.com/hashicorp/consul/pull/20306

This PR starts the HCP manger when an HCP link resource is created instead of when Consul is started. This allows the linking process to be initiated via the HCP link API.

These changes are best viewed commit-by-commit. A summary of the changes are:

- Check explicitly for ACL policies since the HCP manger will create a token, so `acl:write` is also now required in addition to `operator:write`
- Change the HCP manager’s Run method to a Start method, keep track of if its running, and only allow to start once
- Always initialize the required HCP components (i.e., SCADA provider, HCP metrics sink) when Consul starts
- Pass the HCP manager as a dependency to the link controller 
- When a link is created, update the HCP manager with the HCP configs and start it

This PR also introduces a breaking change, though that change actually fixes Consul's behavior to match what we have documented. The [agent telemetry docs](https://developer.hashicorp.com/consul/docs/agent/telemetry) and the [agent configuration docs](https://developer.hashicorp.com/consul/docs/agent/config/config-files#telemetry-disable_hostname) for `telemetry.disable_hostname` both state that by default, the hostname of the Consul agent should prefix gauge-type metrics. However, before this PR, if there were no additional metrics sinks enabled, the hostname prefixing was disabled. Now that we're always enabling the HCP metrics sink, we will now always prefix the gauge-metrics by default.

### Testing & Reproduction steps

Linking via API:
1. Start Consul without a `cloud` configuration
2. Make a PUT request to create a link resource
3. Inspect logs to see that HCP manager has started
4. Check in HCP portal that all features are working as expected (server info and services syncing, observability, global workflows, etc.)
5. Make a GET request to the link and check the status is successful

Other variations tested:
- Linking via configuration rather than the API 
- Bootstrapping via configuration

### Links
- https://hashicorp.atlassian.net/browse/CC-7044

### PR Checklist
